Improve Silo Golshan example (#631)
lethe-cfd/lethe: doc/source/examples/dem/silo/silo.rst

Mesh
~~~~~

> Contrary to previous examples, in this example, we use a mesh generated using Gmsh. The Gmsh (with an extension of .msh) file is located inside the example folder. We refine this mesh once by setting ``initial refinement=1`` to ensure that the cells are sufficiently small to enable an efficient contact detection. ``check diamond cells=true`` enables searching for diamond-shaped boundary cells. Diamond cells are cell which only share a line with the boundary and not a complete face. Such cells can occur in unstructured grids, but they are detrimental to the stability of DEM simualtions. Enabling this option adds these cells to the particle-wall contact search cells and is necessary to ensure stable collisions with the wall in the vicinity of diamond cells.

Reviewer: ```suggestion
Contrary to previous examples, in this example, we use a mesh generated using ``Gmsh <https://gmsh.info/>``_. The Gmsh file (extension .msh) is located inside the example folder. We refine this mesh once by setting ``initial refinement=1`` to ensure that the cells are sufficiently small to enable efficient contact detection. This is important because if we do not apply the refinement, some cells become diamond-shaped. Diamond cells are cells that only share a line with the boundary and not a complete face. ``check diamond cells=true`` enables searching for diamond-shaped boundary cells. Such cells can occur in unstructured grids, but they are detrimental to the stability of DEM simulations. Enabling this option adds these cells to the particle-wall contact search cells and is necessary to ensure stable collisions with the wall in the vicinity of diamond cells.
```

Mesh
~~~~~

> Contrary to previous examples, in this example, we use a mesh generated using ``Gmsh <https://gmsh.info/>``_. The Gmsh file (extension .msh) is located inside the example folder. We refine this mesh once by setting ``initial refinement=1`` to ensure that the cells are sufficiently small to enable efficient contact detection. The mesh generated by Gmsh contains diamond cells are cells that only share a line with the boundary and not a complete face. ``check diamond cells=true`` enables searching for diamond-shaped boundary cells. Such cells can occur in unstructured grids, but they are detrimental to the stability of DEM simulations. Enabling this option adds these cells to the particle-wall contact search cells and is necessary to ensure stable collisions with the wall in the vicinity of diamond cells.

Reviewer: ```suggestion
Contrary to previous examples, in this example, we use a mesh generated using ``Gmsh <https://gmsh.info/>``_. The Gmsh file (extension .msh) is located inside the example folder. We refine this mesh once by setting ``initial refinement=1`` to ensure that the cells are sufficiently small to enable efficient contact detection. The mesh generated by Gmsh contains diamond cells, which are cells that only share a line with the boundary and not a complete face. ``check diamond cells=true`` enables searching for diamond-shaped boundary cells. Such cells can appear in unstructured grids, but they are detrimental to the stability of DEM simulations. Enabling this option adds these cells to the particle-wall contact search cells and is necessary to ensure stable collisions with the wall in the vicinity of diamond cells.
```

  mpirun -np 32 dem_3d silo-Golshan.prm

.. warning::

> 	This example takes approximately 48 hours on 32 cores. This high computational cost is because of the long time of the simulation (40 s) and the high number of particles. To number of particles can be reduced to have a shorter simulation time. For example, the entire simulatio can be ran in less than an hour on an 8 cores processor if only 13 230 particles are inserted.

Reviewer: ```suggestion
	This example takes approximately 48 hours on 32 cores. This high computational cost is due to the long simulation time (40 s of real-time) and the high number of particles. The number of particles can be reduced to have a shorter simulation time. For example, the entire simulation could be run in less than an hour on an 8 cores processor if only 13 230 particles are inserted.
```
